sys/xtimer/xtimer_core.c

```diff
@@ -179,7 +179,7 @@ int _xtimer_set_absolute(xtimer_t *timer, uint32_t target)
     }
 
     unsigned state = disableIRQ();
-    if ( !_this_high_period(target) ) {
+    if ( timer->long_target > _long_cnt || !_this_high_period(target) ) {
```

Comment:
could you add some parentheses around each individual condition for better readability?